server/test/unit/server/webservices/views/test_util.py

```diff
@@ -69,3 +72,96 @@ def test_generate_redirect_response(self, mock_iri_to_uri):
         self.assertEqual(redirect_response._headers['location'][1],
                          str(mock_iri_to_uri.return_value))
         mock_iri_to_uri.assert_called_once_with(href)
+
+
+class TestMustHaveJSONBody(unittest.TestCase):
+
+    def test_must_have_json_body_valid(self):
+        mock_request = mock.MagicMock()
+        mock_request.body = '{"Valid": "JSON"}'
+
+        @must_have_json_body
```

Comment:
I never considered using the decorator to decorate a function and then test it all within a test. neat.